services/venice-controller/src/main/java/com/linkedin/venice/controller/InstanceStatusDecider.java

```diff
@@ -67,7 +67,8 @@ static NodeRemovableResult isRemovable(
     try {
       // If instance is not alive, it's removable.
       if (!HelixUtils.isLiveInstance(clusterName, instanceId, resources.getHelixManager())) {
-        return NodeRemovableResult.removableResult();
+        return NodeRemovableResult
+            .removableResult("Instance " + instanceId + " not found, either dead or wrong instance ID");
```

Comment:
Or we can just say it's not part of the liveInstance for this cluster anymore?

Reply:
Sure